src/lib/parser/default_impl_populator.rs

```diff
@@ -0,0 +1,32 @@
+use std::collections::BTreeMap;
+
+use crate::{ast::visit_mut::*, ast::*};
+
+#[derive(Debug)]
+pub struct  DefaultImplPopulator {
+    pub traits: BTreeMap<String, Trait>,
+}
+
+impl<'a> VisitorMut<'a> for DefaultImplPopulator {
+    fn visit_trait(&mut self, trait_: &mut Trait) {
+        self.traits.insert(trait_.name.get_name(), trait_.clone());
+    }
+
+    fn visit_impl(&mut self, i: &'a mut Impl) {
+        if i.types.is_empty() {
+            return;
+        }
+
+        let trait_name = i.name.get_name();
+        let trait_ = self.traits.get(&trait_name).unwrap();
+
+        i.defs.extend(trait_.default_impl.clone());
```

Comment:
We hard-copy the function declaration into the impl. This is not OK and does not allow for default method overriding in the impl

Reply:
This is a change beyond the scope of this PR, need a new issue for that